core/src/main/scala/org/apache/spark/sql/execution/columnar/JDBCAppendableRelation.scala

```diff
@@ -229,7 +243,7 @@ abstract case class JDBCAppendableRelation(
           JdbcExtendedUtils.executeUpdate(tableStr, conn)
           dialect match {
             case d: JdbcExtendedDialect => d.initializeTable(tableName,
-              sqlContext.conf.caseSensitiveAnalysis, conn)
+              sqlContext.conf.caseSensitiveAnalysis, conn, sysConnFactory)
```

Comment:
Why is sysConnFactory required separately? Shouldn't the "conn" itself have the relevant authorization?